runtime/doc/lua-plugin.txt

```diff
@@ -0,0 +1,432 @@
+*lua-plugin.txt*                     Nvim
+
+                            NVIM REFERENCE MANUAL
+
+		  Guide to developing Lua plugins for Nvim
+
+
+                                       Type |gO| to see the table of contents.
+
+==============================================================================
+Introduction                                                       *lua-plugin*
+
+This is a guide for getting started with Nvim plugin development. It is not
+intended as a set of rules, but as a collection of recommendations for good
+practices.
+
+For a guide to using Lua in Nvim, please refer to |lua-guide|.
+
+==============================================================================
+Type safety                                            *lua-plugin-type-safety*
+
+Lua, as a dynamically typed language, is great for configuration. It provides
+virtually immediate feedback.
+But for larger projects, this can be a double-edged sword, leaving your plugin
+susceptible to unexpected bugs at the wrong time.
+
+You can leverage LuaCATS https://luals.github.io/wiki/annotations/
+annotations, along with lua-language-server https://luals.github.io/ to catch
+potential bugs in your CI before your plugin's users do.
+
+------------------------------------------------------------------------------
+Tools						 *lua-plugin-type-safety-tools*
+
+- lua-typecheck-action https://github.com/marketplace/actions/lua-typecheck-action
```

Comment:
This also seems completely broken. Let's take it easy with the recommendations unless we're certain they're reliable.

Reply:
I maintain this one. How are you testing it?
It has a workflow that runs on itself: https://github.com/mrcjkb/lua-typecheck-action/actions/workflows/typecheck.yml
It just needs a properly configured .luarc.json for packages that have dependencies.

Reply:
I use this and also know a few projects which use it where it works just fine

Reply:
Oh, it defaults to checking `lua` directory which doesn't exist in neovim. Is there a reason we can't just check the actual repo root?

Reply:
OK, you know what? I'll just open 10 issues in your repo and we can work on it there.

Reply:
Right. I assume `llscheck` giving wonky results might also be due to the incorrect `.luarc.json` file?

Reply:
@mrcjkb wanna send a separate PR to neovim to fix `.luarc.json` and `test/.luarc.json`? You seem to already have a good grasp of how these should be configured. (I looked at the docs for luals and got scared)

Reply:
Sure. I'm not quite sure how to fix the `${3rd}` libraries here. That would require [using addons](https://github.com/LuaLS/LLS-Addons). Maybe we should open an issue first?

Reply:
https://github.com/neovim/neovim/pull/29115